Lib/fontTools/merge.py

```diff
@@ -564,6 +572,50 @@ def merge(self, m, tables):
 def mapLookups(self, lookupMap):
 	pass
 
+
+# Gets unicode and font index from the composite gid
+# @param {string} gid - e.g. uni000A#2
+# @return {string} - glyph name
+# @return {int} - font index
+def getGlyphNameAndFontIndex(gid):
+	assert '#' in gid and gid.split('#')[-1].isdigit() and len(gid.split('#')) == 2, 'incorrect gid format'
+	return gid.split('#')[0], int(gid.split('#')[1])
+	
+# Checks if the given glyphs specified by gid are euqal or not.
+# Two glyphs are considered as equal iff:
+# 1. Outlines are equal.
+# 2. Advance width are equal.
+# @param {list<TTFont>} fonts
+# @param {string} gid_0 - e.g. uni0000#1
+# @param {string} gid_1 - e.g. space#2
+# @return {bool}
+def isGlyphSame(fonts, gid_0, gid_1):
+	# Checks outline
+	index_0 = getGlyphNameAndFontIndex(gid_0)[1]
+	index_1 = getGlyphNameAndFontIndex(gid_1)[1]
+	
+	assert fonts[index_0].has_key('glyf') and fonts[index_1].has_key('glyf')
```

Comment:
just a nit: I'd use `'glyf' in fonts[index_0]` instead of `has_key`. Even if both works, since `TTFont` is a user-defined class and not a built-in `dict` which no longer has `has_key` on python3, the latter stinks too much of legacy python. I wonder if we should fire a deprecation warning and eventually get rid of it.